frontend/src/pages/projects/screens/spawner/SpawnerPage.tsx

```diff
@@ -97,10 +97,16 @@ const SpawnerPage: React.FC<SpawnerPageProps> = ({ existingNotebook }) => {
 
   const { size: notebookSize } = useNotebookDeploymentSize(existingNotebook);
   React.useEffect(() => {
+    const customSize: NotebookSize = { name: 'Custom', resources: {} };
     if (notebookSize) {
       setSelectedSize(notebookSize.name);
+    } else {
+      if (existingNotebook && !sizes.some((size) => size.name === 'Custom')) {
+        sizes.unshift(customSize);
+      }
+      setSelectedSize(customSize.name);
```

Comment:
This logic prevents the ability for the customer to use `Custom` as a value name. Why did you do that?

Can we not make our "Custom" significantly more ambitious with an internal id/name that we have a constant for -- and then use that to render special options?

Reply:
This is very weird logic. Please have a test for this.

Example rendering if I set a 'Custom' NotebookSize in the OdhDashboardConfig:
![image](https://github.com/opendatahub-io/odh-dashboard/assets/8126518/4755c4db-0e50-43e1-8bad-8cd51125fb87)